libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -0,0 +1,538 @@
+    }
+
+    @Override
+    public void onMediaUploadSucceeded(final String localMediaId, final MediaFile mediaFile) {
+    }
+
+    @Override
+    public void onMediaUploadProgress(final String localMediaId, final float progress) {
+    }
+
+    @Override
+    public void onMediaUploadFailed(final String localMediaId, final MediaType
+            mediaType, final String errorMessage) {
+    }
+
+    @Override
+    public void onGalleryMediaUploadSucceeded(final long galleryId, long remoteMediaId, int remaining) {
+    }
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        // Toggle action bar auto-hiding for the new orientation
+        if (newConfig.orientation == Configuration.ORIENTATION_LANDSCAPE
+                && !getResources().getBoolean(R.bool.is_large_tablet_landscape)) {
+            mHideActionBarOnSoftKeyboardUp = true;
+            hideActionBarIfNeeded();
+        } else {
+            mHideActionBarOnSoftKeyboardUp = false;
+            showActionBarIfNeeded();
+        }
+    }
+
+    @Override
+    public boolean onTouch(View view, MotionEvent event) {
+        // In landscape mode, if the title or content view has received a touch event, the keyboard will be
+        // displayed and the action bar should hide
+        if (event.getAction() == MotionEvent.ACTION_UP
+                && getResources().getConfiguration().orientation == Configuration.ORIENTATION_LANDSCAPE) {
+            mHideActionBarOnSoftKeyboardUp = true;
+            hideActionBarIfNeeded();
+        }
+        return false;
+    }
+
+    /**
+     * Hide the action bar if needed. Don't hide it if
+     * - a hardware keyboard is connected.
+     * - the soft keyboard is not visible.
+     * - it's not visible.
+     */
+    private void hideActionBarIfNeeded() {
+        ActionBar actionBar = getActionBar();
+        if (actionBar == null) {
+            return;
+        }
+        if (!isHardwareKeyboardPresent()
+                && mHideActionBarOnSoftKeyboardUp
+                && actionBar.isShowing()) {
+            getActionBar().hide();
+        }
+    }
+
+    /**
+     * Show the action bar if needed.
+     */
+    private void showActionBarIfNeeded() {
+        ActionBar actionBar = getActionBar();
+        if (actionBar == null) {
+            return;
+        }
+        if (!actionBar.isShowing()) {
+            actionBar.show();
+        }
+    }
+
+    /**
+     * Returns true if a hardware keyboard is detected, otherwise false.
+     */
+    private boolean isHardwareKeyboardPresent() {
+        Configuration config = getResources().getConfiguration();
+        boolean returnValue = false;
+        if (config.keyboard != Configuration.KEYBOARD_NOKEYS) {
+            returnValue = true;
+        }
+        return returnValue;
+    }
+
+    public boolean onToolbarMediaButtonClicked() {
```

Comment:
We can change the return type to `void`.

```suggestion
    public void onToolbarMediaButtonClicked() {
```